components/provider_ppucd/src/support.rs

```diff
@@ -0,0 +1,111 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/master/LICENSE ).
+
+use crate::parse_ppucd;
+use crate::structs::*;
+use icu_locid::LanguageIdentifier;
+use icu_locid_macros::langid;
+use icu_provider::prelude::*;
+use std::borrow::Cow;
+use std::convert::{TryFrom, TryInto};
+use std::fs::File;
+use std::io::Read;
+use std::marker::PhantomData;
+
+#[derive(Debug)]
+pub struct PpucdDataProvider<'d> {
+    pub ppucd_props: UnicodeProperties,
+    _phantom: PhantomData<&'d ()>, // placeholder for when we need the lifetime param
+}
+
+impl<'d> PpucdDataProvider<'d> {
+    pub fn new(prop_str: &str) -> Self {
+        let data: UnicodeProperties = parse_ppucd::parse(prop_str.to_string());
+        PpucdDataProvider {
+            ppucd_props: data,
+            _phantom: PhantomData,
+        }
+    }
+
+    pub fn from_prop(ppucd_prop: UnicodeProperty) -> Self {
+        PpucdDataProvider {
+            ppucd_props: UnicodeProperties {
+                props: vec![ppucd_prop],
+            },
+            _phantom: PhantomData,
+        }
+    }
+}
+
+impl<'d> DataProvider<'d> for PpucdDataProvider<'d> {
+    fn load(&self, req: &DataRequest) -> Result<DataResponse<'d>, DataError> {
+        const UND: LanguageIdentifier = langid!("und");
+        let data_key: &DataKey = &req.data_key;
+        let data_key_str: &str = data_key.sub_category.as_str();
+        let props_data: &UnicodeProperties = &self.ppucd_props;
+        let matching_prop: Option<&UnicodeProperty> =
+            props_data.props.iter().find(|p| p.name == data_key_str);
```

Comment:
I'd probably use a different data structure for UnicodeProperties since you are forced to do this linear-time lookup.  Either a HashMap or a (sorted) TupleVecMap.